src/lib/dialog/dialog-ref.ts

```diff
@@ -14,10 +14,18 @@ export class MdDialogRef<T> {
   /** The instance of component opened into the dialog. */
   componentInstance: T;
 
+  /** Expose overlay backdrop click event */
+  backdropClicked: Observable<void>;
+
+  /** Subject for notifying the user that esc key way pressed */
+  escapePressed = new Subject<void>();
```

Comment:
What could I differentiate? The problem is that `afterClosed()` triggers only when `close` is called, which actually [closes](https://github.com/angular/material2/blob/master/src/lib/dialog/dialog-ref.ts#L28) the dialog . I need to be able to catch any close **attempt** while `disableClose` is `true` and let user know about it, so he can modify the state himself. Would you be happier is backdrop and escape were merged into one observable, say `closeAttempt`?